src/datasources/location-data-source.ts

```diff
@@ -0,0 +1,30 @@
+import { openmrsFetch, OpenmrsResource } from '@openmrs/esm-framework';
+import { DataSource } from '../api/types';
+
+export class LocationDataSource implements DataSource<OpenmrsResource> {
```

Comment:
Why doesn't this class extend the base data source?

```suggestion
export class LocationDataSource extends BaseOpenMRSDataSource {
```

That way you don't have to implement the `toUuidAndDisplay()`